pkg/instance/instance.go

```diff
@@ -24,3 +32,54 @@ func retrieveByName(cl *clientset.Clientset, ns, name string) (*v1beta1.ServiceI
 	}
 	return instance, nil
 }
+
+func provision(cl *clientset.Clientset, namespace, instanceName, className, planName string, params map[string]string,
+) (*v1beta1.ServiceInstance, error) {
+	paramsJson, err := json.Marshal(params)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal the request parameters %v (%s)", params, err)
+	}
+
+	request := &v1beta1.ServiceInstance{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      instanceName,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.ServiceInstanceSpec{
+			PlanReference: v1beta1.PlanReference{
+				ClusterServiceClassExternalName: className,
+				ClusterServicePlanExternalName:  planName,
+			},
+			Parameters: &runtime.RawExtension{
+				Raw: paramsJson,
+			},
+		},
+	}
+
+	result, err := cl.ServicecatalogV1beta1().ServiceInstances(namespace).Create(request)
+	if err != nil {
+		return nil, fmt.Errorf("provision request failed (%s)", err)
+	}
+	return result, nil
+}
+
+// waitForStatus to be populated on an instance.
+// The result of some operations only include a subset of the instance fields, notably missing status.
+func waitForStatus(cl *clientset.Clientset, instance *v1beta1.ServiceInstance) (*v1beta1.ServiceInstance, error) {
+	var err error
+
+	for i := 0; i < waitForStatusRetries; i++ {
+		if len(instance.Status.Conditions) > 0 {
+			break
+		}
+
+		time.Sleep(waitForStatusPollInterval)
+
+		instance, err = retrieveByName(cl, instance.Namespace, instance.Name)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
might be a good idea to do a backoff here. if the service-catalog controller has a long queue, the status won't show up as quickly as one would hope.

also/or, is it worth making the number retries and wait interval configurable as a parameter?

Reply:
Agreed a backoff vs. regular polling would be better. I'll work on that.

Thinking more about the behavior of all the action commands (bind/provision/etc), what do you think about this?

* `svcat provision` does not wait by default. The status will be empty in the output. Useful when you want to chain multiple commands and wait for them later.
* `svcat provision [--wait=true] [--timeout=5m]` waits for the instance to be provisioned or fail. Useful for simple scripts that aren't performing concurrent tasks.
* `svcat get instance NAME --watch` mimics the functionality of `kubectl`. Useful for those who didn't use `--wait`.